packages/lit-dev-content/site/blog/2023-10-04-lit-3.0.md

```diff
@@ -0,0 +1,132 @@
+
+## Even Faster Rendering with the new Lit Template Compiler { #compiler }
+
+The Lit team is excited to announce a new Labs package @lit-labs/compiler, which provides a [TypeScript Transformer](https://github.com/itsdouges/typescript-transformer-handbook#the-basics) that can be run over your JavaScript or TypeScript files to optimize away some of the work that Lit would have to do at runtime!
+
+![Lit Compiler benchmarks](/images/blog/3.0-launch/compiler-benchmarks.png)
+
+In the best case scenario with a template heavy page, we measured a 46% faster first render, and a 21% faster update!
+
+To try out @lit-labs/compiler today, you’ll need a build step that accepts a TypeScript transformer. For Rollup.js users, this could be @rollup/plugin-typescript. An example rollup.config.js  file might look like:
+
+```js
+// File: rollup.config.js
+import typescript from '@rollup/plugin-typescript';
+import {compileLitTemplates} from '@lit-labs/compiler';
+
+export default {
+  // ...
+  plugins: [
+    typescript({
+      transformers: {
+        before: [compileLitTemplates()],
+      },
+    }),
+    // other rollup plugins
+  ],
+};
+```
+
+### What does the transform do?
+
+Given some source code containing an `html`` tag function to declare templates:
+
+```ts
+const hi = (name) => html`<h1>Hello ${name}!</h1>`;
+```
+
+The Lit template transform will remove the html tag function and replace it with something like the following:
+
+```ts
+const b = (s) => s;
+const lit_template_1 = {h: b`<h1>Hello <?></h1>`, parts: [{type: 2, index: 1}]};
+const hi = (name) => ({_$litType$: lit_template_1, values: [name]});
+```
+
+Then when Lit renders the compiled template, it can skip an internal render phase that we’ve called the Prepare phase, which means you get a quicker initial render.
+
+As you can see in the above example, there is some additional code generated as part of the transform. We’ve measured that minified and compressed, you may get a 5% increase in file size for the compiled file. This is something we have plans to address.
+
+### Looking forward
+
+We’d love to hear from you and get feedback on your experience using the transform, as well as hear what you’d like to see optimized! Leave that feedback in this [Labs Feedback discussion](https://github.com/lit/lit/discussions/4117). We’d also like to learn more about what build systems Lit is used in, and welcome contributions!
+
+This is just the beginning. With this new package we have a foundation for layering on additional build-time optimizations. Some optimizations we’ve thought about:
+
+- For a Lit app which can be completely compiled, we could vend an import of lit-html that is smaller.
+- Add an option to the compiler transform to also minify and compress the HTML in the templates.
+- Provide domain specific and targeted transforms, such as a standard decorator transform with a smaller emit.
+- Compress the emitted output file by applying domain specific file compression.
+
```

Comment:
Add dot point mentioning decorator transform that addresses Standard Decorator emit size.

TODO: me